doc/langref.html.in

```diff
@@ -9463,9 +9463,92 @@ test "assert in release fast mode" {
-      <p>TODO: example of building a zig executable</p>
-      <p>TODO: example of building a C library</p>
+
+      <p>Zig's build system is intended to replace existing tools
+      like Make and CMake. It introduces no dependencies, so Zig code
+      can be built anywhere there's a Zig toolchain.</p>
+
+      {#header_open|Building a Zig Executable#}
+      <p>This <code>build.zig</code> file is automatically generated by <code>zig
+      init-exe</code>.</p>
+      {#code_begin|syntax|build#}
+const Builder = @import("std").build.Builder;
+
+pub fn build(b: *Builder) void {
+    // Standard target options allows the person running `zig build` to choose
+    // what target to build for. Here we do not override the defaults, which
+    // means any target is allowed, and the default is native. Other options
+    // for restricting supported target set are available.
+    const target = b.standardTargetOptions(.{});
+
+    // Standard release options allow the person running `zig build` to select
+    // between Debug, ReleaseSafe, ReleaseFast, and ReleaseSmall.
+    const mode = b.standardReleaseOptions();
+
+    // This line tells the Zig build system where to find the file
+    // that contains main and what to call the executable.
+    const exe = b.addExecutable("main", "src/main.zig");
+    exe.setTarget(target);
+    exe.setBuildMode(mode);
+    exe.install();
+
+    const run_cmd = exe.run();
+    run_cmd.step.dependOn(b.getInstallStep());
+
+    // This will be executed by "zig build run"
+    const run_step = b.step("run", "Run the app");
+    run_step.dependOn(&run_cmd.step);
+}
+      {#code_end#}{#header_close#}
+
+      {#header_open|Building a C library#}
+      {#code_begin|syntax#}
+      const Builder = @import("std").build.Builder;
+
+      pub fn build(b: *Builder) void {
+          const mode = b.standardReleaseOptions();
+          // Add a target that generates libbadmath, with no Zig source files.
+          const lib = b.addStaticLibrary("badmath", null);
+          lib.setBuildMode(mode);
+          // This particular library exists entirely in src/lib.c.
+          lib.addCSourceFile("src/lib.c", &[_][]const u8{
+              "-Wall",
+              "-Wextra",
+              "-Werror",
+          });
+          // libbadmath.a will be put in this directory, instead of only
+          // living in zig-cache.
+          lib.setOutputDir("obj");
+          lib.install();
+      }
+      {#code_end#}
+      {#header_close#}
+
+      {#header_open|Extending a C library#}
+      {#code_begin|syntax#}
+const Builder = @import("std").build.Builder;
+
+pub fn build(b: *Builder) void {
+    const mode = b.standardReleaseOptions();
+    // This line tells the build system to make a static library
+    // called "add" using source from "src/main.zig".
+    const lib = b.addStaticLibrary("add", "src/main.zig");
+    lib.setBuildMode(mode);
+    lib.emit_h = true; // Auto-generate header file
+    lib.force_pic = true; // -fPIE
```

Comment:
PIE stands for Position Independent Executable which is not relevant here, and also `-fPIE` is not a flag recognized by Zig. The docs should explain what it does directly rather than rely on C compiler CLI flag knowledge.

```suggestion
    lib.force_pic = true;
```

Reply:
I was trying to provide a hint for the C linker error people will get without that line, but you make a good point about not depending on knowledge from C.